apps/builder/app/routes/rest.project.$projectId.tsx

```diff
@@ -0,0 +1,58 @@
+import { json, type LoaderArgs } from "@remix-run/node";
+import type { Data } from "@webstudio-is/react-sdk";
+import { db as projectDb } from "@webstudio-is/project/index.server";
+import { sentryException } from "~/shared/sentry";
+import { loadProductionCanvasData } from "~/shared/db";
+import { createContext } from "~/shared/context.server";
+import { getUserById, type User } from "~/shared/db/user.server";
+
+export const loader = async ({
+  params,
+  request,
+}: LoaderArgs): Promise<
+  Data & { user: { email: User["email"] } | undefined } & {
+    projectDomain: string;
+  }
+> => {
+  try {
+    const projectId = params.projectId;
+
+    if (projectId === undefined) {
+      throw json("Required project id", { status: 400 });
+    }
+
+    const context = await createContext(request);
+
+    const project = await projectDb.project.loadById(projectId, context);
+    const buildId = project.latestBuild?.buildId ?? undefined;
+
+    if (buildId === undefined) {
```

Comment:
And since now we have same logic as rest/build endpoint
https://github.com/webstudio-is/webstudio-builder/blob/381b5ae480e9c7ce564d33b8e7a4b36130870c33/apps/builder/app/routes/rest.build.$buildId.tsx#L18
And will always have, plz reuse

Reply:
How can I reuse the same code if params.buildId does not exist on this endpoint? The only param is the projectId.

Reply:
Since you read buildId its same code

Reply:
It does not make sense to reuse the code for me as most of the functions are already shared. The sequence is different in this endpoint.

Reply:
Nothing different, existing endppoint is function of buildId, all you need to read buildId here and use same logic as existing in case if we will need to add anything this will be in a single place

Reply:
Based on our discussion, updated the endpoint.